zcash_primitives/src/transaction/tests.rs

```diff
@@ -0,0 +1,5294 @@
+use super::{
+    components::{Amount, Script},
+    sighash::signature_hash,
+    Transaction,
+};
+
+#[test]
```

Comment:
It'd be nice to have a reference from where these test vectors come from. (I think it's from the raw data in sighash tests in zcashd, right?)

Reply:
IIRC the transaction below is one I extracted from a testnet block. I should have commented to that effect, and will do so.

The test vectors further down have references to the scripts used to synthesise them.